src/routes/transactions/helpers/imitation-transactions.helper.ts

```diff
@@ -0,0 +1,104 @@
+import { Inject } from '@nestjs/common';
+
+export class ImitationTransactionsHelper {
+  private readonly vanityAddressChars: number;
+
+  constructor(
+    @Inject(IConfigurationService) configurationService: IConfigurationService,
+  ) {
+    this.vanityAddressChars = configurationService.getOrThrow(
+      'mappings.imitationTransactions.vanityAddressChars',
+    );
+  }
+
+  /**
+   * Filters out outgoing ERC20 transfers that imitate their direct predecessor in
+   * value and vanity (but not the exact) address of the recipient.
+   *
+   * @param transactions - list of transactions to filter
+   * @param previousTransaction - transaction to compare last {@link transactions} against
+   *
+   * Note: this only handles singular imitation transfers. It does not handle multiple
+   * imitation transfers in a row, nor does it compare batched multiSend transactions
+   * as the "distance" between those batched and their imitation may not be immediate.
+   */
+  filterOutgoingErc20ImitationTransfers(
+    transactions: Array<TransactionItem>,
+    previousTransaction: TransactionItem | undefined,
+  ): Array<TransactionItem> {
+    // TODO: Instead of filtering, mark transactions so client can display them differently
+    return transactions.filter((item, i, arr) => {
+      // Executed by Safe - cannot be imitation
+      if (item.transaction.executionInfo) {
+        return item;
+      }
+
+      const prevItem = i === arr.length - 1 ? previousTransaction : arr[i + 1];
+
+      // No reference transaction to filter against
+      if (!prevItem) {
+        return item;
+      }
+
+      const txInfo = item.transaction.txInfo;
+      const prevTxInfo = prevItem.transaction.txInfo;
+
+      if (
+        // Only consider transfers...
+        !isTransferTransactionInfo(txInfo) ||
+        !isTransferTransactionInfo(prevTxInfo) ||
+        // ...of ERC20s...
+        !isErc20Transfer(txInfo.transferInfo) ||
+        !isErc20Transfer(prevTxInfo.transferInfo)
+      ) {
+        return item;
+      }
+
+      // ...that are outgoing
+      const isOutgoing = txInfo.direction === TransferDirection.Outgoing;
+      const isPrevOutgoing =
+        prevTxInfo.direction === TransferDirection.Outgoing;
+      if (!isOutgoing || !isPrevOutgoing) {
+        return item;
+      }
+
+      // Imitation transfers are of the same value...
+      const isSameValue =
+        txInfo.transferInfo.value === prevTxInfo.transferInfo.value;
+      if (!isSameValue) {
+        return item;
+      }
+
+      // ...from vanity (but not exact) recipient address
+      const isSameRecipient =
+        txInfo.recipient.value === prevTxInfo.recipient.value;
+      if (isSameRecipient) {
+        return item;
+      }
+      return !this.isVanityAddress(
+        txInfo.recipient.value,
+        prevTxInfo.recipient.value,
+      );
+    });
+  }
+
+  private isVanityAddress(address1: string, address2: string): boolean {
```

Comment:
Not sure "vanity" here represents the nature of this scam addresses. A better name would be an "imitator address" perhaps?

Reply:
Updated in 84a9cbb13848efab9c180485e2383b95fe9e0e99.